pandas/core/groupby/grouper.py

```diff
@@ -374,7 +374,8 @@ def _set_grouper(self, obj: FrameOrSeries, sort: bool = False):
         # possibly sort
         if (self.sort or sort) and not ax.is_monotonic:
             # use stable sort to support first, last, nth
-            indexer = self.indexer = ax.argsort(kind="mergesort")
+            # TODO: why does putting na_position="first" fix datetimelike cases?
+            indexer = self.indexer = ax.argsort(kind="mergesort", na_position="first")
```

Comment:
```
   def argsort(
        self,
        ascending: bool = True,
        kind: str = "quicksort",
        na_position: str = "last",
 ```
in arrays/base.py so this is strange

Reply:
i think it must involve the fact that on older numpys, dt64.argsort used to put NaT at the beginning instead of the end